skyportal/handlers/api/annotation.py

```diff
@@ -0,0 +1,263 @@
+from marshmallow.exceptions import ValidationError
+from baselayer.app.access import permissions, auth_or_token
+from ..base import BaseHandler
+from ...models import DBSession, Source, Annotation, Group, Candidate, Filter
+
+
+class AnnotationHandler(BaseHandler):
+    @auth_or_token
+    def get(self, annotation_id):
+        """
+        ---
+        description: Retrieve an annotation
+        parameters:
+          - in: path
+            name: annotation_id
+            required: true
+            schema:
+              type: integer
+        responses:
+          200:
+            content:
+              application/json:
+                schema: SingleAnnotation
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        annotation = Annotation.get_if_owned_by(annotation_id, self.current_user)
+        if annotation is None:
+            return self.error('Invalid annotation ID.')
+        return self.success(data=annotation)
+
+    @permissions(['Annotation'])
+    def post(self):
+        """
+        ---
+        description: Post an annotation
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  obj_id:
+                    type: string
+                  origin:
+                     type: string
+                  data:
+                    type: object
+                  group_ids:
+                    type: array
+                    items:
+                      type: integer
+                    description: |
+                      List of group IDs corresponding to which groups should be
+                      able to view comment. Defaults to all of requesting user's
+                      groups.
+
+                required:
+                  - obj_id
+                  - origin
+                  - data
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            annotation_id:
+                              type: integer
+                              description: New annotation ID
+        """
+        data = self.get_json()
+        obj_id = data.get("obj_id")
+        origin = data.get("origin")
+
+        if obj_id is None:
+            return self.error("Missing required field `obj_id`")
+
+        if origin is None or origin == '':
+            return self.error('Missing required field `origin`')
+
+        annotation_data = data.get("data")
```

Comment:
This needs to be validated

Reply:
I.e., need to validate that `data.get('data')` returns valid JSON

Reply:
I need to check that I get a python dictionary?

Reply:
not just any python dictionary, but valid JSON. if you use a marshmallow schema this validation will be handled automatically for you.